manual/contributing.rst

```diff
@@ -20,11 +20,10 @@ https://github.com/qpdf/qpdf/discussions.
 Code Formatting
 ---------------
 
-The qpdf source code is formatting using clang-format ≥ version 15
+The qpdf source code is formatting using clang-format version 16
```

Comment:
Let's leave this as ≥ 16 here and below.

Reply:
If we stick with >= 16, should we add a note saying how to fix format-code if it picks up the wrong version of clang-format?

Reply:
Yeah, I think so. I think we can just say that if a newer version of clang-format produces different results, then fix on a more recent version of clang-format by bumping the minimum required version in the docs and the format-code script. Did you have something different in mind? Since both contributing.rst and format-code have this, I'd probably put a comment in each place pointing to the other place.

Reply:
Let me know if you want me to take this. It's probably just a couple of lines of bash. I can probably knock it out in a couple of minutes if you're less comfortable with bash. (I don't know your comfort level with it.)

Reply:
I'm going to push something up.

Reply:
> I don't know your comfort level with it

zero comfort level.

Reply:
See #991

Reply:
> Did you have something different in mind?

What I had in mind was a comment along the lines of "If 'format-code' picks the wrong version of clang-tidy or cannot find clang-tidy, change 'clang-tidy' to 'clang-tidy-xx' where 'xx' is the version installed on your system".

Reply:
Okay, feel free to suggest any tweaks to the documentation to reflect that. Are we going to rely completely on the IDE for clang-tidy or have a .clang-tidy file (assuming there is such a thing)? I have yet to read up on clang-tidy.

Reply:
CLion uses the .clang-tidy file (if so configured).

I tend to use CLion to reformat code not checked into git and the script as a final safety check. I find this avoids the problem of accidental reflowing of strings and comments when I don't terminate blocks correctly.